ip-whitelist/cmd/resource/resource.go

```diff
@@ -0,0 +1,247 @@
+package resource
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/encoding"
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
+	"github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+)
+
+// Create handles the Create event from the Cloudformation service.
+func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey.Value(), *currentModel.ApiKeys.PrivateKey.Value())
+	if err != nil {
+		return handler.ProgressEvent{}, err
+	}
+
+	projectID := *currentModel.ProjectId.Value()
+
+	request := getProjectIPWhitelistRequest(currentModel)
+
+	_, _, err = client.ProjectIPWhitelist.Create(context.Background(), projectID, request)
+	if err != nil {
+		return handler.ProgressEvent{}, err
+	}
+
+	var withelist []string
+	whiteListMap(request, func(entry string) {
+		withelist = append(withelist, entry)
+	})
+
+	id := encodeStateID(map[string]string{
+		"project_id": projectID,
+		"entries":    strings.Join(withelist, ","),
+	})
+
+	currentModel.Id = encoding.NewString(id)
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Create Complete",
+		ResourceModel:   currentModel,
+	}, nil
+}
+
+func getProjectIPWhitelistRequest(model *Model) []*mongodbatlas.ProjectIPWhitelist {
+	var whitelist []*mongodbatlas.ProjectIPWhitelist
+	for _, w := range model.Whitelist {
+		wl := &mongodbatlas.ProjectIPWhitelist{}
+		if w.Comment != nil {
+			wl.Comment = *w.Comment.Value()
+		}
+		if w.CidrBlock != nil {
+			wl.CIDRBlock = *w.CidrBlock.Value()
+		}
+		if w.IpAddress != nil {
+			wl.IPAddress = *w.IpAddress.Value()
+		}
+
+		whitelist = append(whitelist, wl)
+
+	}
+	return whitelist
+}
+
+func whiteListMap(whitelist []*mongodbatlas.ProjectIPWhitelist, f func(string)) {
+	for _, entry := range whitelist {
+		if entry.CIDRBlock != "" {
+			f(entry.CIDRBlock)
+		} else if entry.IPAddress != "" {
+			f(entry.IPAddress)
+		}
```

Comment:
@themantissa I think we added support for AWS security groups in https://github.com/mongodb/go-client-mongodb-atlas/pull/35 do we want to do the same here?

Reply:
Good call - Yes, definitely need that for this work!

Reply:
This hasn't been released yet right? I think I can add it if we do a release, I could just fetch master for the build but I think this is not desirable.

Reply:
Correct, not yet.  I've been waiting on asking for a release because the IP Whitelist resource change will be a breaking change and the AWS security groups is included in that.  But actually thinking about it we may be able to cut a release for the go client sooner since the Terraform provider should be pinned to a previous version.

Reply:
@marinsalinas can you do a release of the client? Is this ok @themantissa?

Reply:
Okay with me if okay with Marin.  I'm okay since it won't yet impact Terraform until the next release, unless someone is building from master and if so then the risk of this sort of issue should be known.

Reply:
I added the support for AWS Security group but I need the peering to test this